river/Cursor.zig

```diff
@@ -802,26 +935,92 @@ fn processMotion(self: *Self, device: *wlr.InputDevice, time: u32, delta_x: f64,
             data.delta_y = dy - @trunc(dy);
 
             const border_width = if (data.view.draw_borders) server.config.border_width else 0;
-
-            // Set width/height of view, clamp to view size constraints and output dimensions
             const box = &data.view.pending.box;
-            box.width = @intCast(u32, math.max(0, @intCast(i32, box.width) + @floatToInt(i32, dx)));
-            box.height = @intCast(u32, math.max(0, @intCast(i32, box.height) + @floatToInt(i32, dy)));
+            const constraints = data.view.getConstraints();
+
+            // Avoid potential integer overflow. No reasonable client will have
+            // a minimum size even remotely as large, so this should not be
+            // interfering with well behaved toplevels, but it prevents bad view
+            // constraints from crashing us. A max size larger than this is
+            // pretty common however as most views do not have a max size, which
+            // means they have max_width = max_height = maxInt(u32).
+            const max = math.maxInt(i32) / 2;
```

Comment:
Perhaps this logic belongs in `XdgToplevel.getConstraints()` and `XwaylandView.getConstraints()`?

Reply:
Not sure, since it depends on the type of width and height in the constraints sruct.

Perhaps `getConstraints()` should have a comptime type argument so that either i32 or u32 can be requested. Then that logic can be added to the i32 case.